torch/csrc/jit/fuser/kernel_cache.cpp

```diff
@@ -40,6 +46,20 @@ at::optional<KernelSpec*> retrieve(const int64_t key) {
   return &(it->second);
 }
 
+// XXX: This is O(n) where n = # of key-value pairs in the kernel cache.
+// Maybe we should make this average O(1) by adding a graph-to-key cache.
```

Comment:
We definitely should have this cache. Otherwise it will become really slow on huge models that have a lot of fused kernels (and it gets called for every one, even if they're duplicated!!).